suite-native/app/e2e/tests/deeplinkPopup.test.ts

```diff
@@ -0,0 +1,165 @@
+let server: http.Server | undefined;
+
+const openUriScheme = (url: string, platformToOpen: 'android') => {
+    const command = `npx uri-scheme open '${url.replace(/'/g, '')}' --${platformToOpen}`;
+
+    exec(command, (err, stdout, stderr) => {
+        if (err) {
+            console.error(err);
+
+            return;
+        }
+        // eslint-disable-next-line no-console
+        console.info(stdout);
+        console.error(stderr);
+    });
+};
+
+const reverseServerPort = (port: number) => {
+    return new Promise((resolve, reject) => {
+        exec(
+            `/usr/local/bin/platform-tools/adb reverse tcp:${port} tcp:${port}`,
+            (err, stdout, stderr) => {
+                if (err) {
+                    console.error(err);
+
+                    return reject(err);
+                }
+                // eslint-disable-next-line no-console
+                console.info(stdout);
+                console.error(stderr);
+                resolve(stdout);
+            },
+        );
+    });
+};
+
+describe('Deeplink connect popup.', () => {
+    beforeAll(async () => {
+        await new Promise(resolve => {
+            server = http.createServer((req, res) => {
+                if (req.url) {
+                    const url = new URL(req.url, SERVER_URL);
+                    TrezorConnect.handleDeeplink(url.href);
+                    res.statusCode = 200;
+                    res.setHeader('Content-Type', 'text/plain');
+                    res.end('Callback URL received successfully!\n');
+                }
+            });
+
+            server.listen(SERVER_PORT, 'localhost', () => {
+                // eslint-disable-next-line no-console
+                console.info(`Server running at ${SERVER_URL}`);
+                resolve(null);
+            });
+        });
+
+        await prepareTrezorEmulator();
+        await openApp({ newInstance: true });
+        await onOnboarding.finishOnboarding();
+
+        await onCoinEnablingInit.waitForScreen();
+        await onCoinEnablingInit.enableNetwork('regtest');
+        await onCoinEnablingInit.clickOnConfirmButton();
+
+        await waitFor(element(by.id('skip-view-only-mode')))
+            .toBeVisible()
+            .withTimeout(10000); // communication between connected Trezor and app takes some time.
+
+        await element(by.id('skip-view-only-mode')).tap();
+
+        // This `TrezorConnect` instance here is pretending to be the integrator or @trezor/connect-mobile
+        await TrezorConnect.init({
+            manifest: {
+                email: 'developer@xyz.com',
+                appUrl: 'http://your.application.com',
+            },
+            deeplinkOpen: url => {
+                openUriScheme(url, 'android');
+            },
+            deeplinkCallbackUrl: `${SERVER_URL}/connect/`,
+            connectSrc: 'https://dev.suite.sldev.cz/connect/develop/',
+        });
+    });
+
+    beforeEach(async () => {
+        await prepareTrezorEmulator();
```

Comment:
is not Trezor emulator already "prepared" from the  `beforeAll` step? Is it needed here again?

Reply:
I got that "inspiration" from https://github.com/trezor/trezor-suite/blob/develop/suite-native/app/e2e/tests/send.test.ts#L72 but if it not necessary let's get rid of it.

Reply:
Try to remove it. IMO, it should not be needed here. In case it does not work, feel free to put it back.

Reply:
Yes, you are right, it was not required. Removed.